From 9da1c90d8a072ad778da0612418a1729cf2ce3cb Mon Sep 17 00:00:00 2001
From: Andrei Horodniceanu <a.horodniceanu@proton.me>
Date: Mon, 29 Apr 2024 08:40:20 +0300
Subject: [PATCH] meson.build: add `-allinst` when compiling, #2210

This is needed to work around a bug in dmd regarding template
emissions.

Bug: https://github.com/gnunn1/tilix/issues/2210
Bug: https://issues.dlang.org/show_bug.cgi?id=20668

Signed-off-by: Andrei Horodniceanu <a.horodniceanu@proton.me>
---
 meson.build | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index 73f3ebe8..4b02225d 100644
--- a/meson.build
+++ b/meson.build
@@ -5,12 +5,17 @@ project(
     meson_version: '>= 0.56'
 )
 
+# As a workaround for https://issues.dlang.org/show_bug.cgi?id=20668
+# add -allinst
 compiler = meson.get_compiler('d')
 if compiler.get_id() == 'llvm'
-  d_extra_args = ['-vcolumns']
+  d_extra_args = ['-allinst', '-vcolumns']
   d_link_args = []
-else
-  d_extra_args = []
+elif compiler.get_id() == 'dmd'
+  d_extra_args = ['-allinst']
+  d_link_args = []
+elif compiler.get_id() == 'gcc'
+  d_extra_args = ['-fall-instantiations']
   d_link_args = []
 endif
 
-- 
2.47.0

